spec/lib/rmt/cli/smt_importer_spec.rb

```diff
@@ -0,0 +1,285 @@
+    end
+
+    let(:product) { create :product }
+
+    context 'with systems' do
+      let(:system_1) { create :system }
+      let(:system_2) { create :system }
+      let(:system_3) { create :system }
+
+      let(:activations) do
+        [
+          [system_1.login, product.id],
+          [system_2.login, product.id],
+          [system_3.login, product.id]
+        ]
+      end
+
+      it 'creates the activations for the systems' do
+        expect { script.import_activations }.to output(<<-OUTPUT.strip_heredoc).to_stdout
+          Imported activation of #{product.id} for #{system_1.login}
+          Imported activation of #{product.id} for #{system_2.login}
+          Imported activation of #{product.id} for #{system_3.login}
+        OUTPUT
+        expect(Activation.find_by(system: system_1, service: product.service)).not_to be_nil
+        expect(Activation.find_by(system: system_2, service: product.service)).not_to be_nil
+        expect(Activation.find_by(system: system_3, service: product.service)).not_to be_nil
+      end
+    end
+
+    context 'without a system' do
+      let(:activations) { [['NOT_EXISTING', product.id]] }
+
+      it 'does not create a activation for a not existing system' do
+        expect(Activation).not_to receive(:create)
+        expect { script.import_activations }.to output(<<-OUTPUT.strip_heredoc).to_stderr
+          System NOT_EXISTING not found
+        OUTPUT
+      end
+    end
+
+    context 'without a product' do
+      let(:system) { create :system }
+      let(:activations) { [[system.login, 0]] }
+
+      it 'does not create a activation for a not existing system' do
+        expect(Activation).not_to receive(:create)
+        expect { script.import_activations }.to output(<<-OUTPUT.strip_heredoc).to_stderr
+          Product 0 not found
+        OUTPUT
+      end
+    end
+  end
+
+  describe '#import_hardware_info' do
+    it 'warns the user about missing systems to add the hw_info to' do
+      expect { script.import_hardware_info }.to output(<<-OUTPUT.strip_heredoc).to_stderr
+        System SCC_620a989bc9fb44cc95e7f96390bd979e not found
+        System SCC_82660967a72c4c6f9567cd2382b232b9 not found
+      OUTPUT
+    end
+
+    it 'reads all hardware info and creates or updates the info accordingly' do
+      # The systems has to be created before attaching hw_info to it
+      create :system, login: 'SCC_620a989bc9fb44cc95e7f96390bd979e'
+      create :system, login: 'SCC_82660967a72c4c6f9567cd2382b232b9'
+
+      expect { script.import_hardware_info }.to output(<<-OUTPUT.strip_heredoc).to_stdout
+        Hardware information stored for system SCC_620a989bc9fb44cc95e7f96390bd979e
+        Hardware information stored for system SCC_82660967a72c4c6f9567cd2382b232b9
+      OUTPUT
+    end
+  end
+
+
+  # rubocop:disable RSpec/MultipleExpectations
+  describe '#run' do
+    it 'runs all steps including importing the systems' do
```

Comment:
Needs contexts, e.g.: `context 'without --no-system-flag'`. Otherwise looks weird in the output:
```
  #run
    runs all steps including importing the systems
    imports only the repositories and not the systems
```